packages/block-library/src/navigation/edit/index.js

```diff
@@ -301,6 +296,18 @@ function Navigation( {
 	] = useState();
 	const [ detectedOverlayColor, setDetectedOverlayColor ] = useState();
 
+	// "placeholder" shown if:
+	// - we don't have uncontrolled blocks.
+	// - (legacy) we have a Navigation Area without a ref attribute pointing to a Navigation Post.
+	// - we don't have a ref attribute pointing to a Navigation Post.
+	const isPlaceholderShown =
+		!! ( ! hasUncontrolledInnerBlocks || isWithinUnassignedArea ) && ! ref;
+
+	// "loading" state:
+	// - there is a ref attribute pointing to a Navigation Post
+	// - the Navigation Post hasn't resolved yet.
+	const isLoading = !! ( ref && ! isEntityAvailable );
```

Comment:
What if we omit the `!!` and just let the `if ( isLoading ) {` do the coercion implicitly?

Reply:
I think in this case we don't need explicit coercion.

Reply:
My point was that there is no escaping the coercion. Whether in the variable declaration using `!!` or in an `if` condition, it is happening. If you want to the variable to be a boolean from the start, then how about something like `typeof ref === "string" && ! isEntityAvailable`  or `ref !== null && ! isEntityAvailable` or even `!! ref && ! isEntityAvailable` without the wrapping parenthesis?

Reply:
>  I prefer not to rely on implicit coercion because it requires the developer reading the code to have to think about it.

Does moving the coercion from the `if` to `const` removes the need to think about it?